multisig_wallet/transaction.py

```diff
@@ -97,48 +73,61 @@ def description(self) -> str:
     def to_dict(self):
         tx_dict = self.__dict__
         tx_dict['_destination'] = str(self.destination)
-        del tx_dict['_flexible_struct_format']
         return tx_dict
 
-    @staticmethod
-    def from_bytes(buf: bytes):
-        transaction_executed, destination, value, flexible_struct_format_len = \
-            unpack(Transaction._fixed_struct_format, buf[:FIXED_STRUCT_SIZE])
-
-        flexible_struct_format = \
-            buf[FIXED_STRUCT_SIZE: FIXED_STRUCT_SIZE + flexible_struct_format_len].decode()
-        method, params, description = unpack(flexible_struct_format,
-                                             buf[FIXED_STRUCT_SIZE + flexible_struct_format_len:])
+    @classmethod
+    def create_transaction_with_validation(cls,
+                                           destination: Address,
+                                           method: str,
+                                           params: str,
+                                           value: int,
+                                           description: str,
+                                           executed: bool = False):
+        # as None type can't be converted to bytes, must be changed to ""
+        method = "" if method is None else method
+        params = "" if params is None else params
 
-        return Transaction(executed=transaction_executed,
-                           destination=Address.from_bytes(destination.strip(b'\x00')),
-                           method=method.decode(encoding="utf-8"),
-                           params=params.decode(encoding="utf-8"),
-                           value=int.from_bytes(value, DATA_BYTE_ORDER),
-                           description=description.decode(encoding="utf-8"))
+        if len(method) > MAX_METHOD_LEN \
+                or len(params) > MAX_PARAMS_LEN \
+                or len(description) > MAX_DESCRIPTION_LEN:
+            revert("too long parameter length")
+        try:
+            value.to_bytes(DEFAULT_VALUE_BYTES, DATA_BYTE_ORDER)
+        except OverflowError:
+            revert("exceed ICX amount you can send at one time")
+
+        return cls(executed=executed,
+                   destination=destination,
+                   value=value,
+                   method=method,
+                   params=params,
+                   description=description)
+
+    @classmethod
+    def from_bytes(cls, buf: bytes):
+        encoded_executed = buf[0]
```

Comment:
To avoid warning message.
```suggestion
        encoded_executed = bool(buf[0])
```